src/components/modal/addSrtModal/AddSrtModal.tsx

```diff
@@ -0,0 +1,290 @@
+import { Select } from '../../select/Select';
+import { useState, useEffect } from 'react';
+import { Button } from '../../button/Button';
+import { SrtSource } from '../../../interfaces/Source';
+import styles from './AddSrtModal.module.css';
+import { Loader } from '../../loader/Loader';
+import { useIngests } from '../../../hooks/ingests';
+import Input from '../../input/Input';
+
+type AddSrtModalProps = {
+  open: boolean;
+  loading: boolean;
+  onAbort: () => void;
+  onConfirm: (ingestUuid: string, srtPayload: SrtSource) => void;
+};
+
+type SelectOptions = 'Caller' | 'Listener';
+
+export function AddSrtModal({
+  open,
+  loading,
+  onAbort,
+  onConfirm
+}: AddSrtModalProps) {
+  const ingests = useIngests();
+
+  const [ingestUuid, setIngestUuid] = useState<string>('');
+  const [ingestName, setIngestName] = useState<string>('');
+  const [mode, setMode] = useState<SelectOptions>('Listener');
+  const [localIp, setLocalIp] = useState<string>('0.0.0.0');
+  const [localPort, setLocalPort] = useState<number>(1234);
+  const [remoteIp, setRemoteIp] = useState<string>('127.0.0.1');
+  const [remotePort, setRemotePort] = useState<number>(1234);
+  const [latency, setLatency] = useState<number>(120);
+  const [name, setName] = useState<string>('My SRT source');
+  const [passphrase, setPassphrase] = useState<string>();
+  const [isNameError, setIsNameError] = useState<boolean>(false);
+  const [isIngestNameError, setIsIngestNameError] = useState<boolean>(false);
+  const [srtPayload, setSrtPayload] = useState<SrtSource>({
+    latency_ms: latency,
+    local_ip: localIp,
+    local_port: localPort,
+    mode: 'listener',
+    name: name,
+    passphrase: passphrase,
+    remote_ip: remoteIp,
+    remote_port: remotePort
+  });
+
+  useEffect(() => {
+    if (ingestName !== '') {
+      const selectedIngest = ingests.find(
+        (ingest) =>
+          ingest.name.toLowerCase() === ingestName.toLowerCase().trim()
+      );
+
+      if (selectedIngest) {
+        setIngestUuid(selectedIngest.uuid);
+      }
+    }
+  }, [ingestName, ingests]);
+
+  useEffect(() => {
+    setSrtPayload({
+      latency_ms: latency || undefined,
+      local_ip: localIp || undefined,
+      local_port: localPort || undefined,
+      mode: mode === 'Listener' ? 'listener' : 'caller',
+      name: name,
+      passphrase: passphrase || undefined,
+      remote_ip: mode === 'Caller' ? remoteIp : undefined,
+      remote_port: mode === 'Caller' ? remotePort : undefined
+    });
+  }, [
+    mode,
+    localIp,
+    localPort,
+    remoteIp,
+    remotePort,
+    latency,
+    name,
+    passphrase
+  ]);
+
+  useEffect(() => {
+    if (isNameError) {
+      if (name.length !== 0 || name !== '') {
+        setIsNameError(false);
+      }
+    }
+  }, [isNameError]);
+
+  const handleCloseModal = () => {
+    setIsNameError(false);
+    setIsIngestNameError(false);
+    onAbort();
+  };
+
+  const handleCreateSrtSource = () => {
```

Comment:
There would probably need to be a check for port as well, I have now accidentily created a srt without a port. Redid it and checked, it gets port: undefined, not sure thats desirable.

Reply:
Yes, right now I just did the fields that are mandatory in the API request mandatory (listener/caller and name) but it makes sense to make all of them required really in the GUI